sklearn/metrics/_scorer.py

```diff
@@ -955,10 +955,11 @@ def get_scorer_names():
             None,
         ],
         "allow_none": ["boolean"],
+        "raise_exc": ["boolean"],
     },
     prefer_skip_nested_validation=True,
 )
-def check_scoring(estimator=None, scoring=None, *, allow_none=False):
+def check_scoring(estimator=None, scoring=None, *, allow_none=False, raise_exc=True):
```

Comment:
Could we also add an example of the multimetric in the `Examples` section.

Reply:
Yes, I also think this would be great. I have made such an example, but since we store the whole traceback with `format_exc()` within `_MultimetricScorer` as a string, I am sure that doctest would not be so content about that.

Here is what it looks like for me:

```python
    >>> from sklearn.metrics import make_scorer, accuracy_score, mean_squared_log_error
    >>> X, y = load_iris(return_X_y=True)
    >>> y *= -1
    >>> clf = DecisionTreeClassifier().fit(X, y)
    >>> scoring = {
    >>>     "accuracy": make_scorer(accuracy_score),
    >>>     "mean_squared_log_error": make_scorer(mean_squared_log_error),
    >>> }
    >>> scoring_call = check_scoring(estimator=clf, scoring=scoring, raise_exc=False)
    >>> scores = scoring_call(clf, X, y)
    >>> scores
    {'accuracy': 1.0,
    'mean_squared_log_error': 'Traceback (most recent call last):\n  File "/home/stefanie/code/scikit-learn_dev/scikit-learn/sklearn/metrics/_scorer.py", line 139, in __call__\n    score = scorer._score(\n            ^^^^^^^^^^^^^^\n  File "/home/stefanie/code/scikit-learn_dev/scikit-learn/sklearn/metrics/_scorer.py", line 376, in _score\n    return self._sign * self._score_func(y_true, y_pred, **scoring_kwargs)\n                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/home/stefanie/code/scikit-learn_dev/scikit-learn/sklearn/utils/_param_validation.py", line 213, in wrapper\n    return func(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^\n  File "/home/stefanie/code/scikit-learn_dev/scikit-learn/sklearn/metrics/_regression.py", line 695, in mean_squared_log_error\n    raise ValueError(\nValueError: Mean Squared Logarithmic Error cannot be used when targets contain negative values.\n'}
```
I have put a more generic variation of this and asked doctest to exclude it. Would this be a good way to deal with this?